conformance/utils/suite/experimental_profiles.go

```diff
@@ -0,0 +1,164 @@
+//go:build experimental
+// +build experimental
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suite
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// -----------------------------------------------------------------------------
+// Conformance Profiles - Public Types
+// -----------------------------------------------------------------------------
+
+// ConformanceProfile is a group of features that have a related purpose, e.g.
+// to cover specific protocol support or a specific feature present in Gateway
+// API.
+//
+// For more details see the relevant GEP: https://gateway-api.sigs.k8s.io/geps/gep-1709/
+type ConformanceProfile struct {
+	Name             ConformanceProfileName
+	CoreFeatures     sets.Set[SupportedFeature]
+	ExtendedFeatures sets.Set[SupportedFeature]
+}
+
+type ConformanceProfileName string
+
+const (
+	// HTTPConformanceProfileName indicates the name of the conformance profile
+	// which covers HTTP functionality, such as the HTTPRoute API.
+	HTTPConformanceProfileName ConformanceProfileName = "HTTP"
+
+	// TLSConformanceProfileName indicates the name of the conformance profile
+	// which covers TLS stream functionality, such as the TLSRoute API.
+	TLSConformanceProfileName ConformanceProfileName = "TLS"
+
+	// MeshConformanceProfileName indicates the name of the conformance profile
+	// which covers service mesh functionality.
+	MeshConformanceProfileName ConformanceProfileName = "MESH"
+)
+
+// -----------------------------------------------------------------------------
+// Conformance Profiles - Public Vars
+// -----------------------------------------------------------------------------
+
+var (
+	// HTTPConformanceProfile is a ConformanceProfile that covers testing HTTP
+	// related functionality with Gateways.
+	HTTPConformanceProfile = ConformanceProfile{
+		Name: HTTPConformanceProfileName,
+		CoreFeatures: sets.New(
+			SupportGateway,
+			SupportReferenceGrant,
+			SupportHTTPRoute,
+		),
+		ExtendedFeatures: sets.New(
+			SupportHTTPRouteQueryParamMatching,
```

Comment:
can we reuse https://github.com/kubernetes-sigs/gateway-api/blob/387072cb29001a3771ff5e20ed5fde990749c6a6/conformance/utils/suite/features.go#L145 ?

Reply:
Yes, it makes sense! I re-used that set here https://github.com/kubernetes-sigs/gateway-api/pull/1967/commits/7b7933191a43b5d6c404954958fa89997ed3b37a. I needed to slightly change the original sets as the original set therein declared was adding the HTTP core features as part of the extended features.